### PR TITLE
Add support for IOT batch endpoint

### PR DESCRIPTION
**Pull request by branchvincent**
## Why?
* Support for making batched requests to IOT API

## What changed?
- [x] Add endpoint and models for IOT batched request
- [x] Add `IotService.get_time_series_for_fields` method to batch time series requests
- [x] Add simple test case

### CODE REVIEW

File: contxt/services/iot.py [hunk 1]
```diff
@@ -114,6 +125,79 @@ def get_time_series_for_field(
             per_page=per_page,
         )
 
+    def get_time_series_for_fields(
```
Comment (bretcj7) [inline contxt/services/iot.py]: Why not combine get_time_series_for_field and this new method to just take a list of field[s]?

Or keep the method signature the same for the singular method and just call the new method so you don't split logic?

Comment (branchvincent) [reply contxt/services/iot.py]: I could, but the behavior and return type is pretty different in those cases.

1. `get_time_series_for_field`: returns one paginated time series (i.e. the time series is lazily loaded)
2. `get_time_series_for_field`: returns a list of non-paginated time series (i.e. the pages are loaded up front via batch requests)

The caller may prefer the behavior of one over the other depending on what they need to do.

Comment (bretcj7) [reply contxt/services/iot.py]: Ok, i just wanted to see if there was a way to consolidate.  Will the caller be confused with the very similar names or do you think the pydoc will ensure they are calling what is needed?

**Comment by bretcj7** [comment]
Why is this off the 'feat/poetry'?  Are you spawning a feature from a feature?

File: contxt/services/iot.py [hunk 2]
```diff
@@ -114,6 +125,79 @@ def get_time_series_for_field(
...
+
+            if not any_success:
+                raise IOError(f"All {len(requests)} batched requests to IOT API failed")
+
+        fields_by_name = {f.field_human_name: f for f in fields}
+        return [
+            FieldTimeSeries(field=fields_by_name[name], time_series=series)
+            for name, series in records.items()
+        ]
+
```
Comment (bretcj7) [inline contxt/services/iot.py]: Just a suggestion, but does using something like async/await from asyncio make sense to use here instead of the queueing?

**Pull request merged**